hsclient/json_models.py

```diff
@@ -34,24 +34,24 @@ def handle_empty_website(cls, v):
 
 
 class ResourcePreview(BaseModel):
-    resource_type: str = None
-    resource_title: str = None
-    resource_id: str = None
-    abstract: str = None
+    resource_type: str
+    resource_title: str
+    resource_id: str
+    abstract: Optional[str] = None
     authors: List[str] = []
-    creator: str = None
-    doi: str = None
-    date_created: str = None
-    date_last_updated: str = None
-    public: bool = None
-    discoverable: bool = None
-    shareable: bool = None
-    coverages: Dict[str, str] = None
-    immutable: bool = None
-    published: bool = None
-    resource_url: str = None
-    resource_map_url: str = None
-    resource_metadata_url: str = None
+    creator: str
+    doi: Optional[str] = None
+    date_created: str
+    date_last_updated: str
+    public: bool
+    discoverable: bool
+    shareable: bool
+    coverages: Union[List[Dict[str, Any]], List] = []
```

Comment:
I'm surprised by this

Reply:
I think since the `search` api (https://github.com/hydroshare/hsclient/blob/master/hsclient/hydroshare.py#L1522) has a parameter to search on spatial coverage (which is currently blocked), the preview model has the coverages field.

Reply:
thanks